### PR TITLE
Support ASPNETCORE_URLS to set server urls

### PR DESCRIPTION
**Pull request by davidfowl**
/cc @Tratcher @glennc

### CODE REVIEW

**Comment by Tratcher** [comment]
#755

File: src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs [hunk 1]
```diff
@@ -47,14 +46,8 @@ public WebHostBuilder()
...
-            if (Environment.GetEnvironmentVariable("Hosting:Environment") != null)
-            {
-                Console.WriteLine("The environment variable 'Hosting:Environment' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
-            }
-            if (Environment.GetEnvironmentVariable("ASPNET_ENV") != null)
-            {
-                Console.WriteLine("The environment variable 'ASPNET_ENV' is obsolete and has been replaced with 'ASPNETCORE_ENVIRONMENT'");
-            }
+            // Add the default server.urls key
+            UseSetting(WebHostDefaults.ServerUrlsKey, Environment.GetEnvironmentVariable("ASPNETCORE_URLS"));
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: I thought `urls` was going to become the primary key over here: https://github.com/aspnet/Hosting/blob/abe6dd5692c0d8bda573a25cdeb708a908dce517/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L191
And then `server.urls` would become the obsolete fallback.

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: That would break existing `hosting.json` files with server.urls.

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: You can read both in WebHost

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: Why? That's just more complicated

Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs]: Made the change

**Pull request merged**